libndt7/src/main/java/net/measurementlab/ndt/android/Client.java

```diff
@@ -0,0 +1,251 @@
+package net.measurementlab.ndt.android;
+
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.cert.X509Certificate;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.WebSocket;
+import okhttp3.WebSocketListener;
+import okio.ByteString;
+
+// TODO(bassosimone): do we need locking for this class?
+
+public class Client extends WebSocketListener {
+
+    @NonNull
+    private final Settings settings;
+
+    private double count = 0.0;
+    private boolean rv = true;
+    private long t0 = 0;
+    private long tLast = 0;
+    private final long measurementInterval = TimeUnit.NANOSECONDS.convert(250, TimeUnit.MILLISECONDS);
+
+    @NonNull
+    private static final String TAG = "Client";
+
+    public Client(@NonNull Settings settings) {
+        this.settings = settings;
+    }
+
+    public void onLogInfo(@Nullable String message) {
+        Log.d(TAG, "onLogInfo: " + message);
+        // NOTHING
```

Comment:
I think this `// NOTHING` comment can be removed since now there's a log message. I probably forgot to remove this comment when I added the log. Can you perhaps add this change as part of this PR?